src/mnemonic.rs

```diff
@@ -29,18 +29,21 @@ use crypto::{
 use error::Error;
 use rand::{thread_rng, RngCore};
 use std::io::Cursor;
+use std::fmt::{Display, Formatter, Error as FmtError};
 
 #[derive(Clone, Eq, PartialEq, Debug)]
 pub struct Mnemonic(Vec<usize>);
 
-impl ToString for Mnemonic {
-    fn to_string(&self) -> String {
-        self.0
+impl Display for Mnemonic {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), FmtError> {
+        let seed_stirng = self.0
```

Comment:
Could you try implement this without allocations? That should be quite straightforward. Just iterate over the words and `write!(` them one by one with a space in between.

Reply:
Sure!